prereise/gather/griddata/transmission/geometry.py

```diff
@@ -0,0 +1,272 @@
+                return self.calculate_equivalent_spacing_circular(conductor_distance)
+            if self.layout == "flat":
+                return self.calculate_equivalent_spacing_flat(conductor_distance)
+            raise ValueError(f"Unknown layout: {self.layout}")
+
+    def calculate_equivalent_spacing_circular(self, conductor_distance):
+        if self.n == 3:
+            return (conductor_distance * self.spacing ** 2) ** (1 / 3)
+        if self.n == 4:
+            return (conductor_distance * self.spacing ** 3 * 2 ** (1 / 2)) ** (1 / 4)
+        raise NotImplementedError("Geometry calculations not implemented for n > 4")
+
+    def calculate_equivalent_spacing_flat(self, conductor_distance):
+        if self.n == 3:
+            return (conductor_distance * 2 * self.spacing ** 2) ** (1 / 3)
+        if self.n == 4:
+            return (conductor_distance * 12 * self.spacing ** 3) ** (1 / 8)
+        raise NotImplementedError("Geometry calculations not implemented for n > 4")
+
+
+@dataclass
+class PhaseLocations:
+    """Represent the locations of each conductor bundle on a transmission tower.
+
+    :param tuple a: the (x, y) location of the single 'A' phase conductor if
+        ``circuits`` == 1, or the ((x1, y1), (x2, y2), ...) locations of the 'A' phase
+        conductors if ``circuits`` > 1. Units are meters.
+    :param tuple a: the (x, y) location of the single 'B' phase conductor if
+        ``circuits`` == 1, or the ((x1, y1), (x2, y2), ...) locations of the 'B' phase
+        conductors if ``circuits`` > 1. Units are meters.
+    :param tuple a: the (x, y) location of the single 'C' phase conductor if
+        ``circuits`` == 1, or the ((x1, y1), (x2, y2), ...) locations of the 'C' phase
+        conductors if ``circuits`` > 1. Units are meters.
+    :param int circuits: the number of circuits on the tower.
+    """
+
+    a: tuple
+    b: tuple
+    c: tuple
+    circuits: int = 1
+
+    def __post_init__(self):
+        if not (len(self.a) == len(self.b) == len(self.c)):
+            raise ValueError("each phase location must have the same length")
+
+
+@dataclass
+class Tower:
+    """Given the geometry of a transmission tower and conductor bundle information,
+    estimate per-kilometer inductance, resistance, and shunt capacitance.
+
+    :param PhaseLocations locations: the locations of each conductor bundle.
+    :param ConductorBundle bundle: the locations of each conductor bundle.
+    """
+
+    locations: PhaseLocations
+    bundle: ConductorBundle
+
+    def __post_init__(self):
+        self.a = self.locations.a
+        self.b = self.locations.b
+        self.c = self.locations.c
+        self.calculate_distances()
+        self.resistance = self.bundle.resistance_per_km / self.locations.circuits
+        self.inductance = self.calculate_inductance_per_km()
+        self.capacitance = self.calculate_shunt_capacitance_per_km()
+
+    def calculate_distances(self):
+        if self.locations.circuits == 1:
```

Comment:
Would it make sense to have `Tower` subclasses for 1 circut and (either >1 circuit, or specifically 2, not sure)?

Reply:
Another possibility would be moving some calculations to the `PhaseLocations` class if they don't depend on `self.bundle`, in which case the subclasses would extend from there.

Reply:
That could work. You're right that the logic for 2 circuits should work for any plural number of circuits (although it's only been validated against a double circuit example).

The `__post_init__` would be shared among the two designs, but the other methods would differ. Would we be able to design it so that we can still instantiate a `Tower`, and the number of circuits would automatically determine which subclass of object is generated/returned? I guess at that point `Tower` becomes a factory, rather than a class, and looks at the `circuits` input to determine whether to instantiate and return a `SingleCircuitTower` or `MultiCircuitTower` object.

Either that, or maybe we can tweak the format or parsing for the single-circuit inputs, such that the multiple-circuit equations still work with only a single circuit.

Reply:
I think you're right that the distance calculations can live in the `PhaseLocations` class. Calculating the inductance and the capacitance does need bundle info though.

Reply:
I refactored all of the distance calculations into `PhaseLocations` and a small helper function, so calculations in the `Tower` class are common between single-circuit and multi-circuit designs. There are still different calculations between single-circuit and multi-circuit distances within `PhaseLocations`, but they're organized behind self-contained methods and a single logic branch, which should hopefully be clearer to follow.